d2bs/kolbot/libs/bots/BattleOrders.js

```diff
@@ -5,6 +5,43 @@
 */
 
 function BattleOrders() {
+	this.AmTardy = function() {
+		let party;
+
+		if (Config.BattleOrders.SkipIfTardy) {
+			party = getParty();
+
+			AreaInfoLoop:
+			while (true) {
+				if (Misc.getPlayerCount() <= 1) {
+					throw new Error("Empty game"); // Alone in game
+				}
+
+				if (party) {
+					do {
+						if (party.name !== me.name && party.area) {
+							break AreaInfoLoop; // Can read player area
+						}
+					} while (party.getNext());
+				}
+
+				delay(1000);
+			}
+
+			if (party) {
+				do {
+					if (party.area === 131 || party.area === 132 || party.area === 108 || party.area === 39) { // Player is in Throne of Destruction or Worldstone Chamber or Chaos Sanctuary
+						print("ÿc1I'm late to BOs. Moving on...");
+						return true;
```

Comment:
If it detects your party members are at baal/chaos, its a bit sad to see the bot quit the game, if you didnt set up the QuitOnFailed[Get|Give] option. Here we might want to override the quit function?

Reply:
I _think_ I set the `QuitOnFailedWhatever` to False in the default, so this doesn't happen but lemmy refine that a bit